runtime/plugins/org.jboss.tools.runtime.core/src/org/jboss/tools/runtime/core/model/AbstractRuntimeDetectorDelegate.java

```diff
@@ -22,11 +24,32 @@
 public abstract class AbstractRuntimeDetectorDelegate implements
 		IRuntimeDetectorDelegate {
 
-	@Override
+	private boolean loggedWarning = false;
+	
+	@Override @Deprecated
 	public void initializeRuntimes(List<RuntimeDefinition> runtimeDefinitions) {
 	}
 
 	@Override
+	public boolean initializeRuntime(RuntimeDefinition runtimeDefinition) throws CoreException {
+		// Provide some migration code, for now.  This code may be removed at any time in the future. 
+		
+		// Since this is a new method, and some clients might not be implementing it, we 
+		// should redirect to the original method initializeRuntimes.
+		initializeRuntimes(Collections.singletonList(runtimeDefinition));
+		
+		// Log an error / warning indicating other handlers should update their API call
```

Comment:
this is wrong. the error will be logged when calling initializeRuntime(RuntimeDefinition runtimeDefinition) . you have a chicken/egg issue here :-)